### PR TITLE
Build Starboard Operator based on Red Hat Universal Base Image 8 Minimal

### PR DESCRIPTION
**Pull request by prakhar-aqua**
This PR contains changes in the .goreleaser yaml ,  Makefile and addition of Dockerfile with ubi8-minimal as base Image for OCP Certification of Starboard-Operator.

### CODE REVIEW

File: .goreleaser.yaml [hunk 1]
```diff
@@ -92,16 +93,17 @@ dockers:
       - "--label=org.opencontainers.image.revision={{ .FullCommit }}"
       - "--label=org.opencontainers.image.documentation=https://aquasecurity.github.io/starboard/v{{ .Version }}/"
       - "--platform=linux/amd64"
+  
   - image_templates:
-      - "docker.io/aquasec/starboard-operator:{{ .Version }}-amd64"
+      - "docker.io/aquasec/starboard-operator:ubi8-{{ .Version }}-amd64"
```
Comment (danielpacak) [inline .goreleaser.yaml:98]: Why are you removing Alpine-based image? I think we should keep publishing Alpine-based images and add UBI-based images.

File: build/starboard-operator-ubi8/Dockerfile.ubi8 [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+LABEL product=aquasec
+LABEL com.aquasec.baseimage=registry.access.redhat.com/ubi8/ubi-minimal
+LABEL maintainer="admin@aquasec.com"
+LABEL name="Starboard-Operator ubi8"
+LABEL vendor="Aqua Security Software Ltd."
+LABEL summary="Aqua security Starboard-Operator-ubi8"
+LABEL description="Aqua Security Starboard is a Kubernetes-Native Security toolkit"
+
+RUN microdnf install yum
+RUN yum install bash-completion -y
```
[the remark this answers is not in the document]
Comment (danielpacak) [reply build/starboard-operator-ubi8/Dockerfile.ubi8:14]: But is completion required to add such user really?

[the remark this answers is not in the document]
Comment (prakhar-aqua) [reply build/starboard-operator-ubi8/Dockerfile.ubi8:14]: yes bash-completion is required, I tried locally and it didn't work without it.

File: build/starboard-operator-ubi8/Dockerfile.ubi8 [hunk 3]
```diff
@@ -0,0 +1,21 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal
```
[the remark this answers is not in the document]
Comment (prakhar-aqua) [reply build/starboard-operator-ubi8/Dockerfile.ubi8]: tag added

File: build/starboard-operator-ubi8/Dockerfile.ubi8 [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+MAINTAINER admin@aquasec.com
+LABEL com.aquasec.component=starboard-operator
+LABEL product=aquasec
+LABEL com.aquasec.baseimage=registry.access.redhat.com/ubi8/ubi-minimal
+LABEL maintainer="admin@aquasec.com"
+LABEL name="Starboard-Operator ubi8"
+LABEL vendor="Aqua Security Software Ltd."
+LABEL summary="Aqua security Starboard-Operator-ubi8"
+LABEL description="Aqua Security Starboard is a Kubernetes-Native Security toolkit"
```
[the remark this answers is not in the document]
Comment (prakhar-aqua) [reply build/starboard-operator-ubi8/Dockerfile.ubi8]: removed the labels from Dockerfile

File: .goreleaser.yaml [hunk 5]
```diff
@@ -92,16 +93,17 @@ dockers:
...
     use: buildx
     goos: linux
-    dockerfile: build/starboard-operator/Dockerfile
+    dockerfile: build/starboard-operator-ubi8/Dockerfile.ubi8
     goarch: amd64
     ids:
       - starboard-operator
     build_flag_templates:
-      - "--label=org.opencontainers.image.title=starboard-operator"
+      - "--label=org.opencontainers.image.title=starboard-operator-ubi8"
```
[the remark this answers is not in the document]
Comment (prakhar-aqua) [reply .goreleaser.yaml]: label corrected

File: .goreleaser.yaml [hunk 6]
```diff
@@ -236,6 +277,25 @@ dockers:
       - "--label=org.opencontainers.image.revision={{ .FullCommit }}"
       - "--label=org.opencontainers.image.documentation=https://aquasecurity.github.io/starboard/v{{ .Version }}/"
       - "--platform=linux/s390x"
+
+  - image_templates:
+      - "docker.io/aquasec/starboard-operator:ubi8-{{ .Version }}-ppc64"
+    use: buildx
+    goos: linux
+    dockerfile: build/starboard-operator-ubi8/Dockerfile.ubi8
+    goarch: ppc64
```
[the remark this answers is not in the document]
Comment (prakhar-aqua) [reply .goreleaser.yaml]: arch corrected to ppc64le ( little-endian version of ppc64)

File: .goreleaser.yaml (hunk 1 above)
Comment (danielpacak) [reply .goreleaser.yaml]: That's okey. Alpine images don't have suffixes

File: build/starboard-operator-ubi8/Dockerfile.ubi8 (hunk 2 above)
[the remark this answers is not in the document]
Comment (danielpacak) [reply build/starboard-operator-ubi8/Dockerfile.ubi8]: We agreed that it's not required and we'll only install shadow-utils package. Later on, we could use existing user that comes with the UBI image.

**Review by danielpacak — APPROVED** [review]

File: .goreleaser.yaml [hunk 7]
```diff
@@ -252,3 +326,9 @@ docker_manifests:
       - "aquasec/starboard-scanner-aqua:{{ .Version }}-amd64"
       - "aquasec/starboard-scanner-aqua:{{ .Version }}-arm64"
       - "aquasec/starboard-scanner-aqua:{{ .Version }}-s390x"
+  - name_template: "aquasec/starboard-operator:{{ .Version }}-ubi8"
+    image_templates:
+      - "aquasec/starboard-scanner-aqua:{{ .Version }}-ubi8-amd64"
+      - "aquasec/starboard-scanner-aqua:{{ .Version }}-ubi8-arm64"
+      - "aquasec/starboard-scanner-aqua:{{ .Version }}-ubi8-s390x"
+      - "aquasec/starboard-scanner-aqua:{{ .Version }}-ubi8-ppc64le"
```
Comment (danielpacak) [inline .goreleaser.yaml:334]: -> starboard-operator

**Pull request merged**